apps/dotcom/client/src/tla/providers/TlaProvider.tsx

```diff
@@ -18,43 +30,69 @@ if (!PUBLISHABLE_KEY) {
 }
 
 export function Component() {
+	const handleAppLevelUiEvent = useCallback<TLUiEventHandler>(() => {
+		// todo, implement handling ui events at the application layer
+	}, [])
+
 	return (
 		<ClerkProvider publishableKey={PUBLISHABLE_KEY} afterSignOutUrl="/q">
-			<TlaRootProviders>
-				<SignedInProvider />
-			</TlaRootProviders>
+			<AppStateOrNotProvider>
```

Comment:
fwiw, this is going to be improved in https://github.com/tldraw/tldraw/pull/4683 where I'm just using `TldrawUiContextProvider` instead of trying to recreate it manually (which will be error-prone)